src/storage-client/src/sink.rs

```diff
@@ -386,12 +394,14 @@ pub async fn build_kafka(
     ensure_kafka_topic(
         &admin_client,
         &connection.topic,
+        // TODO: allow users to configure these parameters.
         TopicConfig {
-            partition_count: connection.partition_count,
-            replication_factor: connection.replication_factor,
-            // TODO: allow users to request compaction cleanup policy instead of
-            // retention.
-            cleanup_policy: TopicCleanupPolicy::Retention(connection.retention),
+            partition_count: -1,
+            replication_factor: -1,
+            cleanup_policy: TopicCleanupPolicy::Retention {
+                ms: Some(-1),
```

Comment:
Is this a behavior change? If I followed the code on main correctly we're currently setting the broker defaults when creating the data topic. If we plan to eventually expose this in SQL it would be better to leave this the same so when we run the future migration we can safely set all existing sinks to use the broker default explicitly

Reply:
Eek! Yes, thanks. Total oversight due to the confusing API for this function. Fixed now.